### PR TITLE
Datadog pgx wrapper

### PR DESCRIPTION
**Pull request by stroem**

### CODE REVIEW

File: v2/trace/ddpgx/conn.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+func (o *traceConn) ConnInfo() *pgtype.ConnInfo {
+	return o.conn.ConnInfo()
+}
+
+func (o *traceConn) Begin(ctx context.Context) (pgx.Tx, error) {
+	startTime := time.Now()
+	tx, err := o.conn.Begin(ctx)
+	tryTrace(ctx, startTime, o.serviceName, "pgx", "Begin", nil, err)
```
Comment (lonnblad) [inline v2/trace/ddpgx/conn.go:25]: consider adding constants for "pgx", "Begin", etc.

File: v2/trace/ddpgx/ddpgx.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	Exec(context.Context, string, ...interface{}) (pgconn.CommandTag, error)
+	Query(context.Context, string, ...interface{}) (pgx.Rows, error)
+	QueryRow(context.Context, string, ...interface{}) pgx.Row
+	Close(context.Context) error
+}
+
+func Connect(ctx context.Context, serviceName, url string) (Connection, error) {
+	startTime := time.Now()
+	conn, err := pgx.Connect(ctx, url)
+	tryTrace(ctx, startTime, serviceName, "pgx", "Connect", nil, err)
```
Comment (j-sv) [inline v2/trace/ddpgx/ddpgx.go:25]: Argument order in tryTrace is driver, serviceName, resource.

File: v2/trace/ddpgx/trace.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+	for i := range args {
+		key := fmt.Sprintf("sql.args.%d", i)
+		output[key] = args[i]
+	}
+
+	return output
+}
+
+func tryTrace(ctx context.Context, startTime time.Time, driver, serviceName, resource string, metadata map[string]interface{}, err error) {
```
Comment (lonnblad) [inline v2/trace/ddpgx/trace.go:23]: wrong order: `driver, serviceName, resource` in regards to how the func is used
maybe consider using a struct instead

File: v2/trace/ddpgx/conn.go (hunk 1 above)
Comment (stroem) [reply v2/trace/ddpgx/conn.go]: Updated

File: v2/trace/ddpgx/conn.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgtype"
+	"github.com/jackc/pgx/v4"
+	dd_ext "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+)
+
+type traceConn struct {
+	conn  Connection
+	trace *internalTracer
```
Comment (lonnblad) [inline v2/trace/ddpgx/conn.go:14]: `trace *internalTracer`, why a pointer?

File: v2/trace/ddpgx/conn.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+func (o *traceConn) ConnInfo() *pgtype.ConnInfo {
+	return o.conn.ConnInfo()
+}
+
+func (o *traceConn) Begin(ctx context.Context) (pgx.Tx, error) {
+	o.trace.Start()
```
Comment (lonnblad) [inline v2/trace/ddpgx/conn.go:22]: is this `o.trace.Start()` threadsafe?

**Review by lonnblad — APPROVED** [review]

**Pull request merged**